traffic_ops/app/bin/checks/ToDSCPCheck.pl

```diff
@@ -290,90 +348,92 @@
 
 closelog();
 
-sub get_dscp() {
+sub get_dscp {
    my $url      = shift;
    my $ip       = shift;
    my $dev      = shift;
    my $header   = shift;
    my $protocol = shift;
+   my $port     = shift;
 
    my $tos     = undef;
    my $max_len = 0;
 
    my $src_port = int( rand( 65535 - 1024 ) ) + 1024;
-   TRACE "get_dscp ip:" . $ip . " url:" . $url . " dev:" . $dev . " port:" . $src_port . " ip protocol: ".$protocol;
+   TRACE "TRACE In sub get_dscp";
+   DEBUG "DEBUG get_dscp ip:" . $ip . " url:" . $url . " dev:" . $dev . " port:" . $src_port . " ip protocol: ".$protocol;
 
    # Use curl to get some traffic from the URL, but send the command to the background, so the capture that follows
    # is while traffic is being returned
-   if ($ip =~ m/:/) {
-      TRACE "running ip6";
-      my $curl = "curl --local-port " . $src_port . " --".$protocol." -s ".$url." -H \"Host: ".$header."\" 2>&1 > /dev/null";
-      TRACE "curl: ".$curl;
+   my $curl = "curl --connect-timeout 2 --local-port " . $src_port . " --".$protocol." -s \"".$url."\" -H \"Host: ".$header."\" 2>&1 > /dev/null";
```

Comment:
If you use single quotes instead of double quotes here, you can avoid some of that escaping and improve readability.

Reply:
Fixed.